features/step_definitions/back_office/manage_user_steps.rb

```diff
@@ -0,0 +1,120 @@
+require "pry"
+
+When(/^I invite a new "([^"]*)" user$/) do |user_type|
+  @bo.dashboard_page.govuk_banner.manage_users_link.click
+  expect(@bo.users_page).to be_displayed
+
+  @bo.users_page.invite_user.click
+
+  user_invite_page = @bo.user_invite_page
+
+  expect(user_invite_page).to be_displayed
+
+  @new_user_email = generate_email
+  @new_user_role = user_type
+  element = "#{user_type}_user_radio"
+
+  user_invite_page.email_field.set(@new_user_email)
+  user_invite_page.public_send(element).click
+  user_invite_page.submit_form.click
+
+  expect(@bo.users_page).to be_displayed
+end
+
+Then(/^the new user has the correct back office permissions$/) do
+  look_into_paginated_content_for(@new_user_email)
+
+  within page.first(:css, "tr", text: /.*#{Regexp.quote(@new_user_email)}.*/) do
+    expect(page).to have_text(@new_user_role)
+  end
+end
+
+Then("the new user accepts its invitation and setup a password") do
+  last_email_page = LastEmailPage.new
+  last_email_page.load
+
+  expect(last_email_page.check_email_for_text(["Confirm a waste carriers back office account"])).to be(true)
+  @confirm_waste_carriers_email_link = last_email_page.text.match(/.*href\=\\\"(.*)\\\".*/)[1]
```

Comment:
Will this line always work even if two tests have been run in quick succession? Because the email is spread randomly across two servers, it is possible that an email will load with the subject line "Confirm a waste carriers back office account" but have a link relevant to the user from the previous test that was run.

You might need an extra step to check the email for `@new_user_email` too.

Reply:
Oh right! I will make the `last_email_page.check_email_for_text(["Confirm a waste carriers back office account"])` become `last_email_page.check_email_for_text(["Confirm a waste carriers back office account", @new_user_email])` so we can be sure of that :)

Reply:
Looks good!